crates/kitsune_p2p/fetch/src/pool.rs

```diff
@@ -183,6 +167,8 @@ impl State {
         match self.queue.entry(key) {
             Entry::Vacant(e) => {
                 let sources = if let Some(author) = author {
+                    // TODO This is currently dead code, no callers provide the author. Even if they do, that's the only thing that the
+                    //      `source` can contain. The author field could actually be removed from the input type.
```

Comment:
Leaving this in for the review but I'd like to action it

Reply:
@maackle - I don't really remember what was going on here, but I vaguely remember discussions about future augmentations.

Reply:
I don't remember why or how author is never provided, but the idea is that the author is always a good choice for a candidate to get data from, so we might as well add it in. I think rather the action item should be to make sure the author gets passed along.

Reply:
Oh I see, so even though the `FetchSource` is an enum with only the field `Agent`, that agent could be any agent we know is holding that data but not necessarily the author. I'll update the TODO comment to explain why this isn't covered by tests for now and mention that there is a TODO at one of the call sites about adding author as an input.